lib/manifest-generator.js

```diff
@@ -33,23 +34,70 @@ var ManifestGenerator = Addon.extend({
   postprocessTree: function(type, tree) {
     if (type !== 'all') { return tree; }
 
+    /**
+     * Options we care about:
+     * - `app.env` via Ember CLI
+     * - `app.name` via `root-app/ember-cli-build.js`
+     * - `app.options.assetLoader.noManifest` via `root-app/ember-cli-build.js`
+     * - `app.options.assetLoader.generateURI` via `root-app/ember-cli-build.js`
+     * - `app.options.outputPaths.app.html` via `root-app/ember-cli-build.js`
+     * - `app.project.config(app.env).rootURL` via `root-app/config/environment.js`
+     * - `this.manifestOptions.*` via `var ManifestGenerator = require('manifest-generator');`
+     *   - This file is a base addon which can be extended.
+     *   - Can overwrite anything.
+     *   - Expected to provide:
+     *     - `this.manifestOptions.bundlesLocation`
+     *     - `this.manifestOptions.supportedTypes`
+     *
+     * From these we return early if `noManifest` is set.
+     *
+     * Otherwise we generate:
+     * ```
+     * var manifestOptions = {
+     *   appName,
+     *   bundlesLocation,
+     *   generateURI,
+     *   rootURL,
+     *   supportedTypes
+     * }
+     * ```
+     */
+
     var app = findHost(this);
-    var options = app && app.options;
-    var assetLoaderOptions = (options && options.assetLoader) || {};
+
+    // root's ember-cli-build.js
+    var appBuildtimeOptions = app && app.options;
+    var assetLoaderOptions = (appBuildtimeOptions && appBuildtimeOptions.assetLoader) || {};
 
     if (assetLoaderOptions.noManifest) {
       return tree;
     }
 
+    // root's config/environment.js
+    var runtimeOptions = app.project.config(app.env);
+
+    // Normalize the `rootURL` per Ember CLI's pattern.
+    // Also checks `baseURL` to support the old `baseURL` setting.
```

Comment:
In Ember CLI 2.6 we used `baseURL` and 2.7 we added `rootURL` and deprecated (but still support) `baseURL`-only setups. It's not old enough in my opinion to drop the branching support.